### ISSUE TITLE
Allow all CLI options in config file

### ISSUE THREAD

**Issue by levithomason** [prettier/prettier#2506]
It would be handy to allow all CLI options (except for `--find-config-path` and `--config`) to be set via the config file.  I think the ideal scenario is to have as close to a 1:1 relationship between CLI args and config options as possible.  Config file options would be applied first and CLI args would be assigned over the top, allowing them to override the config file.

Currently, it seems `--list-different` and `--write` cannot be set via the config file.  I'd like to set these conditionally so that in CI I `--list-different` while locally I `--write`.

**prettier.config.js**
```js
const prettierConfig = {
  semi: false,
  printWidth: 110,
  singleQuote: true,
  trailingComma: 'all',
}

if (process.env.CI === 'true') {
  prettierConfig.listDifferent = true
} else {
  prettierConfig.write = true
}

module.exports = prettierConfig
```

**Comment by azz** [comment]
We can't easily do this because we get to support config being passed to the API, as well as the CLI.

I'd recommend having two scripts in your package.json

```
"format": "prettier --write",
"check": "prettier --l"
```

**Comment by levithomason** [comment]
Though this can be done, there is a large list of globs to support in this case.  It would be preferred to not have large duplicate scripts with only a single flag difference between the two.

Could you shed a little more light on what "can't do this easily" means technically?

**Comment by vjeux** [comment]
You can do

```
"format": "npm run check"
```

**Comment by levithomason** [comment]
Does `-l --write` still format but also exit non-zero if there were differences?

**Comment by levithomason** [comment]
I'll close this as it seems the preferred approach is not to support all options via the config file.  The current workaround I'm using is to invoke `prettier` via a node script which sets the appropriate options.

**Comment by azz** [comment]
Can you simplify the globs using `.prettierignore`?

**Comment by levithomason** [comment]
Indeed, however, that didn't seem to be supported in the current release `1.5.3`.  I also tried every version back to `1.4.4.` but had issues with those as well.

Should this not be the case?

**Comment by azz** [comment]
Both .prettierrc and .prettier-ignore are unreleased features. They'll be in the next minor release.

**Comment by levithomason** [comment]
👍 Thanks for the info and updates. I know what it takes to support OSS and I'm very thankful for what you and the rest of the team are putting into this.

**Comment by e-cloud** [comment]
@azz `prettier --write` doesn't work. I think it should find the `.prettierrc` file and do the job. Even if i specify the config path, prettier still output the help message and exit.

**Comment by lydell** [comment]
@e-cloud See #2683. That has been fixed and will be released in 1.6.1!
